server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/AbstractRecipientRewriteTableMailet.java

```diff
@@ -144,7 +144,9 @@ public void service(Mail mail) throws MessagingException {
 
             MailAddress nextMap = new MailAddress(rcpt.asString());
             if (getMailetContext().isLocalServer(nextMap.getDomain())) {
-                localRecipients.add(nextMap);
+                if (getMailetContext().isLocalEmail(nextMap)) {
+                    localRecipients.add(nextMap);
+                }

```

Comment:
I know, but i think this correction should be merged as soon as possible.
And we may cherry-pick this commit, tests will be merged later (as there's a stack of different commits to be merged before that).